src/validation.h

```diff
@@ -596,6 +591,10 @@ class CChainState
     /** Chain parameters for this chainstate */
     const CChainParams& m_params;
 
+    //! Optional mempool that is kept in sync with the chain.
+    //! Only the active chainstate has a mempool.
+    CTxMemPool* m_mempool;
```

Comment:
988cfd8914f12ce87fb6ac6499326e140e840c1d

Instead of making this public, I think you should add a getter like `CoinsTip()` and an initialized like `InitCoinsDB`. Then you can revert what @stratospher asks in https://github.com/bitcoin/bitcoin/pull/23465/commits/988cfd8914f12ce87fb6ac6499326e140e840c1d#r756677595.

Reply:
Done in https://github.com/bitcoin/bitcoin/pull/23465/commits/358f94a28176f766634d9701f1688abfcf761c28 .